### ISSUE TITLE
Can't save Artifactory ServerDetails in Generic-Artifactory Integration

### ISSUE THREAD

**Issue by lijinxia** [jfrog/jenkins-artifactory-plugin#408]
Describe the bug
Can't save Artifactory upload/download server setting in Generic-Artifactory Integration

To Reproduce
Open an exist job which enabled Generic-Artifactory Integration or Configurate a new one, Save the config.

Expected behavior
Setting can be saved

Screenshots
configSubmit Failed Webpage

Versions
- Jenkins Artifactory plugin version: 3.10.3
- Jenkins operating system: 2.270
- Artifactory Version:

Additional context
Jenkins log:
Error while serving https://*******/artifactory_freestyle/configSubmit
org.kohsuke.stapler.WrongTypeException: Got type array but no lister class found for type class org.jfrog.hudson.ServerDetails
	at org.kohsuke.stapler.RequestImpl$TypePair.convertJSON(RequestImpl.java:724)
	at org.kohsuke.stapler.RequestImpl.bindJSON(RequestImpl.java:478)
	at org.kohsuke.stapler.RequestImpl.instantiate(RequestImpl.java:787)
Caused: java.lang.IllegalArgumentException: Failed to convert the specsDeployerDetails parameter of the constructor public org.jfrog.hudson.generic.ArtifactoryGenericConfigurator(org.jfrog.hudson.ServerDetails,org.jfrog.hudson.ServerDetails,org.jfrog.hudson.ServerDetails,org.jfrog.hudson.ServerDetails,org.jfrog.hudson.CredentialsConfig,org.jfrog.hudson.CredentialsConfig,java.lang.String,java.lang.String,java.lang.String,java.lang.String,boolean,org.jfrog.hudson.SpecConfiguration,org.jfrog.hudson.SpecConfiguration,boolean,boolean,org.jfrog.hudson.util.IncludesExcludes,boolean,boolean,boolean,boolean,java.lang.String,java.lang.String,boolean)
	at org.kohsuke.stapler.RequestImpl.instantiate(RequestImpl.java:789)
	at org.kohsuke.stapler.RequestImpl.access$200(RequestImpl.java:83)
	at org.kohsuke.stapler.RequestImpl$TypePair.convertJSON(RequestImpl.java:678)

**Comment by yahavi** [comment]
@lijinxia 
Thanks for reporting this issue!
We are working on a fix. We'll keep you updated.

**Comment by yahavi** [comment]
@lijinxia 
Thanks for reporting this issue.
To improve compatibility with the Jenkins core for the current and future versions, we decided to start removing Javascript code and use the Jenkins Jelly whenever it is possible.

To resolve this issue, in #409 we removed the support for legacy patterns from the UI.
Here is a snapshot that should resolve this issue:
https://oss.jfrog.org/artifactory/oss-snapshot-local/org/jenkins-ci/plugins/artifactory/3.10.x-SNAPSHOT/artifactory-3.10.x-20210105.082253-46.hpi

We'd appreciate your feedback on that.
